src/main/java/spireMapOverhaul/patches/interfacePatches/RewardModifierPatches.java

```diff
@@ -0,0 +1,127 @@
+import com.megacrit.cardcrawl.rooms.AbstractRoom;
+import com.megacrit.cardcrawl.saveAndContinue.SaveAndContinue;
+import com.megacrit.cardcrawl.screens.CombatRewardScreen;
+import com.megacrit.cardcrawl.vfx.FastCardObtainEffect;
+import com.megacrit.cardcrawl.vfx.cardManip.ShowCardAndObtainEffect;
+import javassist.CtBehavior;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zoneInterfaces.IRewardModifyingZone;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class RewardModifierPatches {
+    @SpirePatch2(clz = CombatRewardScreen.class, method = "setupItemReward")
+    public static class AddCardRewardsPatch {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void AddCardRewards(CombatRewardScreen __instance) {
+            AbstractZone zone = Wiz.getCurZone();
+            if (zone instanceof IRewardModifyingZone) {
+                ArrayList<ArrayList<AbstractCard>> additionalCardRewards = ((IRewardModifyingZone)zone).getAdditionalCardRewards();
+                for (ArrayList<AbstractCard> cards : additionalCardRewards) {
+                    RewardItem rewardItem = new RewardItem();
+                    rewardItem.cards = cards;
+                    __instance.rewards.add(rewardItem);
+                }
+            }
+        }
+
+        private static class Locator extends SpireInsertLocator {
+            @Override
+            public int[] Locate(CtBehavior ctMethodToPatch) throws Exception {
+                Matcher matcher = new Matcher.NewExprMatcher(RewardItem.class);
+                int[] lines = LineFinder.findAllInOrder(ctMethodToPatch, matcher);
+                // We only want the first two uses, which are respectively the branch for the Vintage modifier (only
+                // elites and bosses drop cards) and the normal branch. The third use is for Prayer Wheel.
+                // The reason this matters (and why this isn't just a postfix patch) is because want to avoid adding
+                // additional card rewards to fights that normally don't drop rewards, and we don't want to reproduce
+                // the logic that CombatRewardScreen.setupItemRewards has for that.
+                return new int[] { lines[0], lines[1] };
+            }
+        }
+    }
+
+    @SpirePatch2(clz = AbstractRoom.class, method = "update")
+    public static class ModifyRewardsPatch {
+        @SpireInsertPatch(locator = Locator.class)
+        public static void ModifyRewards(AbstractRoom __instance) {
+            AbstractZone zone = Wiz.getCurZone();
+            if (zone instanceof IRewardModifyingZone) {
+                // We don't want to affect uses of CombatRewardScreen that are in non-battle events such as Sensory Stone
+                if (AbstractDungeon.getCurrMapNode() != null && AbstractDungeon.getCurrMapNode().room != null && AbstractDungeon.getCurrMapNode().room.isBattleOver) {
```

Comment:
This design choice is worth noting. The reward modifiers in the spreadsheet are generally associated with combats, but there are some zones where it's ambiguous. E.g. for Tetraminus's Broken Space or Darkglade's Heavenly Clouds, what happens if you get Sensory Stone while in the zone, or purchase Orrery? Do those rewards get the modifier applied?

The answer being "no" is most intuitive to me. However, it wouldn't be too had to change if we wanted to allow this. We'd add a method like `affectNonCombatRewards` and check that here.